dashboard/app/models/concerns/text_to_speech.rb

```diff
@@ -183,7 +183,32 @@ def tts_should_update_short_instructions?
   end
 
   def tts_long_instructions_text
-    tts_long_instructions_override || TextToSpeech.sanitize(long_instructions || "")
+    # Instructions in contained levels are used as TTS instead of the instructions of the containing level
+    tts_long_instructions_override || TextToSpeech.sanitize(tts_for_contained_level || long_instructions || "")
+  end
+
+  def tts_for_contained_level
+    all_instructions = []
+
+    contained_levels.each {|contained| all_instructions.push(contained_level_text(contained))}
+    contained_levels.empty? ? nil : all_instructions * "\n"
```

Comment:
should this be `all_instructions.empty?`?

Reply:
My original intention here was: 'if there are no contained levels, then just return nil', but implementation-wise I agree that checking all_instructions makes more sense.